impeller/tessellator/tessellator.cc

```diff
@@ -28,13 +37,21 @@ static int ToTessWindingRule(FillType fill_type) {
   return TESS_WINDING_ODD;
 }
 
-static void DestroyTessellator(TESStesselator* tessellator) {
-  if (tessellator != nullptr) {
-    ::tessDeleteTess(tessellator);
-  }
-}
+class TessellatorContext {
+ public:
+  explicit TessellatorContext(CTessellator c_tessellator)
+      : c_tessellator_(std::move(c_tessellator)) {}
+
+  ~TessellatorContext() { c_tessellator_.reset(); }
+
+  TESStesselator* tessellator() { return c_tessellator_.get(); }
```

Comment:
It's a bit strange to expose this as a public member.

I wonder if we could just have one `Tessellator`. We could then put it on the `Context` object, and that would give it the lifetime you want without exposing the raw pointer to libtess2's object.

Reply:
So there'd be something like `std::shared_ptr<Tessellator> Context::GetTessellator`.

Reply:
Yeah, that seems better. Which Context object are you referring to?

Reply:
Agreed, that the tessellator is backed by libtess2 is a current implementation detail to the rest of the stack.

Reply:
Refactored this so the Tessellator object is cached.